server/storage/wal/version.go

```diff
@@ -213,6 +214,9 @@ func visitEnumNumber(enum protoreflect.EnumDescriptor, number protoreflect.EnumN
 	if err != nil {
 		return err
 	}
+	if int(number) >= enum.Values().Len() || int(number) < 0 {
+		return errors.New("could not visit EnumNumber")
```

Comment:
It would be better if we can get the invalid `number` included in the error message.